notifications-sw.js

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * UsersManager screenshot tests.
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+var token;
+
+self.addEventListener('install', async function() {
+    self.token = new URL(location).searchParams.get('token');
+});
+
+self.addEventListener('activate', async function() {
+    // Fetch any notifications that have already been generated
+    fetchNotifications();
+
+    // Check periodically for new notifications
+    this.timerId = setInterval(function() {
+        fetchNotifications();
+    }, 30 * 1000);  // Timeout is in ms
```

Comment:
Do I see this right we're fetching for notifications every 30 seconds? Could we do this maybe only once per hour or maybe even only every 3 hours? or 6 hours?

Reply:
Yes, 30 seconds for dev purposes while it's still a WIP, will be changed to something much less frequent once it's ready to go. 1 hour, 4 hours and 6 hours have all been suggested previously...